share_extension/ios/extension_post.js

```diff
@@ -503,21 +506,35 @@ export default class ExtensionPost extends PureComponent {
 
     selectTeam = (team, channel) => {
         this.setState({channel, team});
+
+        // Update the channels for the team
+        Client4.getMyChannels(team.id).then((channels) => {
+            const defaultChannel = channels.find((c) => c.name === General.DEFAULT_CHANNEL && c.team_id === team.id);
+            this.updateChannelsInEntities(channels);
+            if (!channel) {
+                this.setState({channel: defaultChannel});
+            }
+        }).catch((error) => {
+            this.setState({error});
+        });
     };
 
     sendMessage = wrapWithPreventDoubleTap(async () => {
-        const {credentials, onClose} = this.props;
-        const {channel, currentUserId, files, value} = this.state;
+        const {authenticated, onClose} = this.props;
+        const {channel, entities, files, value} = this.state;
+        const {currentUserId} = entities.users;
 
         // If no text and no files do nothing
         if (!value && !files.length) {
             return;
         }
 
-        if (currentUserId && credentials) {
+        if (currentUserId && authenticated) {
             await this.auth();
```

Comment:
Maybe I'm misunderstanding, but is this trying to authenticate the user after we're already authenticated?

Reply:
this is in case EMM is enabled to ask to the passcode or fingerprint